test/services/dangling_account_remover_test.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class DanglingAccountRemoverTest < ActiveSupport::TestCase
+  test 'removes accounts that have no relationships' do
+    FactoryBot.create_list(:account, 30)
+
+    assert_difference('Account.count' => -30) do
+      DanglingAccountRemover.run!
+    end
+  end
+
+  test 'keeps accounts that have relationships' do
```

Comment:
The `DanglingAccountRemover` checks relationships that exist today. If it is going to be used more than once (e.g. in a cron job), then we should ensure that no new relationship is overlooked. Can there be maybe a test with auto-discovery of new Account relationships to prevent that?

Reply:
PG will just fail if we have the foreign keys :thinking: at least it did for me when I tried to do anything ugly